Monika After Story/game/zz_consumables.rpy

```diff
@@ -0,0 +1,1400 @@
+
+            OUT:
+                List of MASConsumable objects which are enabled
+
+            NOTE: enabled is regardless of stock amount
+            """
+            consumables = []
+
+            if store.mas_consumables.TYPE_DRINK in store.mas_consumables.consumable_map:
+                consumables.extend([
+                    drink
+                    for drink in store.mas_consumables.consumable_map[mas_consumables.TYPE_DRINK].values()
+                    if drink.enabled()
+                ])
+
+            if store.mas_consumables.TYPE_FOOD in store.mas_consumables.consumable_map:
+                consumables.extend([
+                    food
+                    for food in store.mas_consumables.consumable_map[mas_consumables.TYPE_FOOD].values()
+                    if food.enabled()
+                ])
+
+            return consumables
+
+        @staticmethod
+        def __getCurrentConsumable(_type):
+            """
+            Gets the current consumable, provided by type
+
+            IN:
+                _type - consumable type to get the current consumable for
+
+            OUT:
+                MASConsumable object representing the current consumable object for the type
+                If there's no consumable out by _type, None is returned
+            """
+            return mas_getConsumable(
+                persistent._mas_current_consumable[_type]["id"]
+            )
+
+        @staticmethod
+        def __checkingLogic(_type, curr_cons, startup):
+            """
+            Generalized logic to check if we should have a consumable
+
+            IN:
+                _type - consumable type
+                curr_cons - current_consumable (of _type)
+                startup - whether or not to perform a startup check
+            """
+            available_cons = MASConsumable._getConsumablesForTime(_type)
+
+            #Verify persist data
+            MASConsumable._validatePersistentData(_type)
+
+            #Wear the acs if we don't have it out for some reason
+            if MASConsumable._isHaving(_type) and not monika_chr.is_wearing_acs(curr_cons.acs):
+                monika_chr.wear_acs_pst(curr_cons.acs)
+
+            #Reset if we're having a consumable we shouldn't be having now and we opened the game after its consume time
+            if (
+                MASConsumable._isHaving(_type)
+                and (
+                    not MASConsumable._isStillCons(_type)
+                    and mas_getCurrSeshStart() > persistent._mas_current_consumable[_type]["consume_time"]
+                )
+            ):
+                MASConsumable._reset(_type)
+
+            #If we have no consumables, then there's no point in doing anything
+            if not available_cons:
+                return
+
+            #If we're currently prepping/having anything, we don't need to do anything else
+            if persistent._mas_current_consumable[_type]["id"] is not None:
```

Comment:
the logic above here seems off. It appears possible to wear an ACS (if `isHaving` but not `is_wearing_acs`), but then have that consumable reset because it shouldn't be consumed at this time (if `_isHaving` and (not `isStillCons` and `consume_time` has passed).

`_reset` removes the ACS, so this would waste time running through the `wear_acs` hooks only to remove it after.

This should be rewritten to avoid wearing the ACS until we know for sure it's consumable will be still active.